web/app/scripts/workspace/workspace-drawing-board.js

```diff
@@ -289,11 +292,8 @@ angular.module('biggraph').directive(
         }
 
         scope.onMouseUpOnBox = function(box, event) {
-          if (box.isDirty || scope.pulledPlug || scope.selection.isActive()) {
-            return;
-          }
           const leftButton = event.button === 0;
-          if (!leftButton || event.ctrlKey || event.shiftKey) {
+          if (mouseMoved || !leftButton || event.ctrlKey || event.shiftKey) {
             return;
           }
```

Comment:
You're right, this deserves an explanation! When you click on a box, we want to open the box popup. A click is basically a mouseup event. But there are some exceptions when we don't want a popup on mouseup. Most importantly, we don't want a popup if this was some sort of dragging event. Maybe we dragged the box, maybe we dragged the arrow between boxes, maybe we dragged a selection rectangle. These are the three checks in the deleted `if`.

I replaced that `if` with a new mechanism that just checks whether the mouse has moved. It's simpler, and it fixes a corner case: When you start dragging a box, `isDirty` is set, to mark this change for saving to the backend. Once it's saved, `isDirty` is cleared. But Playwright is going to click on the box after finishing the movement while it's still marked as dirty! Happens to users too if the network or the backend is slow. This is not an issue with `mouseMoved` — this flag is cleared on mouseup.